sdk/storage/src/clients/storage_account_client.rs

```diff
@@ -0,0 +1,427 @@
+use azure_core::errors::AzureError;
+use azure_core::headers::*;
+use azure_core::HttpClient;
+use azure_core::EMPTY_BODY;
+use http::header::*;
+use http::method::Method;
+use http::request::{Builder, Request};
+use ring::hmac;
+use std::sync::Arc;
+use url::Url;
+
+pub(crate) const HEADER_VERSION: &str = "x-ms-version"; //=> [String] }
+
+pub(crate) const AZURE_VERSION: &str = "2019-07-07";
+//pub(crate) const SAS_VERSION: &str = "2019-02-02";
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum StorageCredentials {
+    Key(String, String),
+    SASToken(Vec<(String, String)>),
+    BearerToken(String),
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum ServiceType {
+    Blob,
+    // Queue,
+    // File,
+    Table,
+}
+
+#[derive(Debug, Clone)]
+pub struct StorageAccountClient {
+    storage_credentials: StorageCredentials,
+    http_client: Arc<Box<dyn HttpClient>>,
+    blob_storage_uri: String,
+    table_storage_uri: String,
+    queue_storage_uri: String,
+    filesystem_uri: String,
+}
+
+fn get_sas_token_parms(sas_token: &str) -> Vec<(String, String)> {
+    Url::options()
+        // Any base url will do: we just need to parse the SAS token
+        // to get its query pairs.
+        .base_url(Some(&Url::parse("https://blob.core.windows.net").unwrap()))
```

Comment:
Is this call to `base_url` actually needed?

Reply:
I believe the intent of the code is to parse a SAS token provided without base URL. It's not cleat though, we should remember to document it in `new_sas_token`.